src/utils/kbn_field_types.js

```diff
@@ -0,0 +1,104 @@
+
+export class KbnFieldType {
+  constructor(options = {}) {
+    const {
+      name,
+      sortable = false,
+      filterable = false,
+      esTypes = []
+    } = options;
+
+    Object.defineProperties(this, {
```

Comment:
A bit meh, but I understand why, so I'm good with it. If this is what we have to do to have immutable objects, this is what we do.